File: Creational/Constructor/swift/constructor.swift
/// Constructors are used for instantiating an object.
/// It initializes the values to the data members of the class 
/// when an object of that class is being created

class Car {
    var model: String
    var year: Int
    var miles: Int


    // MARK: - Constructor

	init(model: String, year: Int, miles: Int) {
		self.model = model
		self.year = year
		self.miles = miles
    }
}
  
// instantiating an object of the class
let obj = Car("Jeep", 2009, 50000)

File: Creational/Constructor/ruby/constructor.rb
# Constructors are used for instantiating an object.
# It initializes the values to the data members of the class
# when an object of that class is being created

class Car
	# Constructor
	def initialize(model, year, miles)
		# Variables prefixed with @ are instance variables
    # which belong to the object being constructed
		@model = model
		@year = year
		@miles = miles
  end

	# Method for printing the object as a string
  # When using methods for printing strings like `puts`
  # this will be automatically used
	def to_s
		"#{@model} has done #{@miles} miles"
  end
end

# Instantiating an object of the class
obj = Car.new("Jeep", 2009, 50000)

# Print the object
puts obj
#=> Jeep has done 50000 miles
